Sources/Configuration/ConfigurationManager.swift

```diff
@@ -18,65 +18,125 @@ import Foundation
 
-    /// List of known deserializers for parsing raw data (i.e., from file or HTTP requests)
+    /// List of known deserializers for parsing raw data (i.e. from file or HTTP requests)
     private var deserializers: [String: Deserializer] = [
         JSONDeserializer.shared.name: JSONDeserializer.shared,
         PLISTDeserializer.shared.name: PLISTDeserializer.shared
     ]
 
-    /// Defaults to `--`
+    /**
+     Prefix used to denote a command line argument as a configuration path-value pair. Defaults to `--`
+    
+     For example: ```./myApp --path.to.configuration=value```
+    
+     Note: This can be set to your preferred prefix when instantiating `ConfigurationManager`.
+     See: `init(commandLineArgumentKeyPrefix:commandLineArgumentPathSeparator:environmentVariablePathSeparator:parseStringToObject:)`
+    */
     public var commandLineArgumentKeyPrefix: String
 
-    /// Defaults to `.`
+    /**
+     Path separator to specify the components of a path that is passed in as a command line argument. Defaults to `.`
+     
+     For example: ```./myApp --path.to.configuration=value```
+     
+     Note: This can be set according to your preferences when instantiating `ConfigurationManager`.
+     See: `init(commandLineArgumentKeyPrefix:commandLineArgumentPathSeparator:environmentVariablePathSeparator:parseStringToObject:)`
+     */
     public var commandLineArgumentPathSeparator: String
 
-    /// Defaults to `__`
+    /**
+     Path separator to specify the components of a path that is passed in as an environment variable. Defaults to `__`
+     
+     For example: ```PATH__TO__CONFIGURATION=value```
+     
+     Note: This can be set according to your preferences when instantiating `ConfigurationManager`.
+     See: `init(commandLineArgumentKeyPrefix:commandLineArgumentPathSeparator:environmentVariablePathSeparator:parseStringToObject:)`
+    */
     public var environmentVariablePathSeparator: String
 
-    /// Defaults to `true`
+    /**
+     Used to indicate if string values in command line arguments and environment variables should be parsed to
+     array or dictionary, if possible, using a known Deserializer. Defaults to `true`
+     
+     Note: This can be set according to your preferences when instantiating `ConfigurationManager`.
+     See: `init(commandLineArgumentKeyPrefix:commandLineArgumentPathSeparator:environmentVariablePathSeparator:parseStringToObject:)`
+    */
     public var parseStringToObject: Bool
 
-    /// Enum to specify configuration source between commandline arguments
-    /// and environment variables.
+    /// Enum to specify the configuration source. The supported options are to load configuration
+    /// data from either commandline arguments or environment variables.
     public enum Source {
-        /// Flag to load configurations from commandline arguments
+        /// Load configurations from commandline arguments.
+        let manager = ConfigurationManager().load(.commandLineArguments)
```

Comment:
How come this line is added as source code? Seems like it's causing compile to fail.